xija/runner.py

```diff
@@ -0,0 +1,212 @@
+            # to descriptor attributes.
+            return self
+        else:
+            return getattr(instance, self.name)
+
+    def __set__(self, instance, value):
+        setattr(instance, self.name, value)
+
+    def __set_name__(self, owner_cls, name):
+        self.name = name
+
+        if not hasattr(owner_cls, 'comp_names'):
+            owner_cls.comp_names = []
+        owner_cls.comp_names.append(name)
+
+
+class ModelRunner:
+    """
+    Class object that knows how to run a particular model.  This
+    needs to be subclassed with specifics of the model components
+    that needs to be set and relevant model values that are output.
+    """
+    def run(self, obs_req, start):
+        """
+        Run the model for the specified ``obs_req``
+
+        This sets an attribute ``final_state``
+        """
+        self.states = self.make_states(obs_req)
+
+        # Now make the model, initialize the states, run, and set ``final_state``
+
+    def make_states(self, obs_req, start):
+        """
+        Make state values required to run the model for ``obs_req``.
+
+        Includes:
+
+        - Generate a maneuver to the new attitude
+        - Assume a SIM translation (if necessary) in the middle of the maneuver
+          (is there a better choice?)
+        - Potentially replicate some of the ORviewer logic for going to zero-FEPs.
+          If this is not done then the answers for ACIS models are conservative.
+        """
+        pass
+
+    def check_against_limits(self):
+        """
+        Check that the computed temperature(s) are within specified limits.
+
+        This could in theory be a tri-state
+        value: OK, not OK, and almost OK (meaning that the worst case violation
+        was within e.g. 20% of the model error).
+
+        :returns: bool (or str?)
+        """
+        pass
+
+
+class AcaModelRunner(ModelRunner):
+    """
+    Specific model components required to run the ACA model.
+
+    Note the limit and model_err values are defaults and could be set
+    directly (from ORviewer) with something like
+    ``AcaModelRunner.aacccdpt.limit = -7.9``.
+    """
+    # These class attributes represent the initial state and required data
+    # values to run the model.
+
+    pitch = ModelComp(float)
+    eclipse = ModelComp(bool)
+    aacccdpt = ModelComp(float, limit=-8.5, model_err=1.0)  # Node
+    aca0 = ModelComp(float)  # Pseudo-node
+
+
+class ObsReqRunner:
+    def __init__(self, ephemeris, eclipse, model_names=DEFAULT_MODELS):
+        """
+        :param ephemeris: dict of ephemeris {time (cxcsec), x, y, z}
+        """
+        self.ephemeris = ephemeris  # needs validation and processing
+        self.obs_reqs = {}
+        self.models = []
+
+        self.model_names = model_names.copy()
+
+        for model_name in model_names:
+            model_runner_name = model_name.capitalize() + 'ModelRunner'
+            self.models[model_name] = globals()[model_runner_name]()
```

Comment:
small syntactical question, are the final parenthesis needed/intended?  I am still building my mental Python parser, but I would think that those parenthesis would cause the variable from `globals()[model_runner_name]` to be evaluated and store the return value, as opposed to storing the value of the variable itself in to the `self.models` dict